### DISCUSSION TITLE
Adding the integration always ends with error

### DISCUSSION THREAD

**Discussion by endrju666** [veista/nilan#59] [Q&A] [answered]
Hi, I bought this modbus converter: https://www.aliexpress.com/item/1005004471558181.html?spm=a2g0o.order_list.order_list_main.5.21ef1802tk1TqC

I can access the modbus converter on local IP 192.168.1.209, but when I try to add the Nilan intergration I always get some error. Either its "Failed to Connect" when I try the port 502 or "Invalid Response from Device" when I try port 80. 

Screenshot of adding integration:
![image](https://user-images.githubusercontent.com/10759379/228290690-2cfbece6-c7ec-462a-8801-7e94824fe3de.png)

The modbus converter settings is like this: 
![image](https://user-images.githubusercontent.com/10759379/228289654-2984c654-4702-4afe-b751-515858544c39.png)

Please do you have any suggestion what to change? Thanks a lot.

**Comment by veista** [comment]
At least your baudrate is wrong.

**Reply by endrju666** [reply]
> At least your baudrate is wrong.

I changed the baud rate to 19200 and parity to even, but same errors.

**Comment by veista** [comment, marked as answer]
Work mode should be server

**Reply by endrju666** [reply]
Awesome, it works. Thank you. 

So these are the steps I did:
- baud rate to 19200
- parity to even
- work mode to TCP server
